app/models/ems_refresh/save_inventory.rb

```diff
@@ -53,6 +53,8 @@ def save_vms_inventory(ems, hashes, target = nil, disconnect = true)
     ]
     remove_keys = child_keys + extra_infra_keys + extra_cloud_keys
 
+    vms_by_ems_ref = ems.vms_and_templates.index_by(&:ems_ref).except(nil)
```

Comment:
Should this be non-archived only or do you want everything (or does vms_and_templates already handle that)?  I'm concerned that if an old vm has an ems_ref that matches (say the case where they update vsphere or whatever), then it will clobber.

Reply:
Using the association off of the ems automatically gets only the non-archived ones (aka the ones with an ems_id)